<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
   "http://www.w3.org/TR/html4/strict.dtd">
<HTML>
   <HEAD>
      <TITLE>My first HTML document</TITLE>
      <style rel="stylesheet" type="text/css">
body {
 font-size: 20px;
 
 margin-top: 50px;
    margin-bottom: 50px;
    margin-right: 80px;
    margin-left: 100px;
    
    padding-top: 50px;
    padding-bottom: 50px;
    padding-right: 80px;
    padding-left: 80px;
    
    line-height:35px;
}
/* img {
 width:900px;
} */
</style>
      <script type="text/x-mathjax-config">
MathJax.Hub.Config({
    "HTML-CSS" : {
        availableFonts : ["STIX"],
        preferredFont : "STIX",
        webFont : "STIX-Web",
        imageFont : null
    }
});
</script>
     <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js" type="text/javascript">    
    MathJax.Hub.Config({
        HTML: ["input/TeX","output/HTML-CSS"],
        TeX: { extensions: ["AMSmath.js","AMSsymbols.js"], 
               equationNumbers: { autoNumber: "AMS" } },
        extensions: ["tex2jax.js"],
        jax: ["input/TeX","output/HTML-CSS"],
        tex2jax: { inlineMath: [ ['$$$','$$$'] ],
                   displayMath: [ ['$$$$','$$$$'] ],
                   processEscapes: true },
        "HTML-CSS": { availableFonts: ["TeX"],
                      linebreaks: { automatic: true } }
    });
</script>
   </HEAD>
   <BODY>
<xmp>
https://www.youtube.com/watch?v=JgF3L9CNw5I
/mnt/1T-5e7/mycodehtml/Biology/SangSooKim/Bioinformatics_basic/003/002_What_is_exome_Sanger_sequencing/main.html

================================================================================
How to find variants

- See video
https://www.youtube.com/watch?v=5VClehvKVw4

- /home/young/Pictures/2019_06_12_13:19:33.png

- Small sequence reads, reference sequence (bottom) as standard genome sequence
/home/young/Pictures/2019_06_12_13:20:18.png

- Small sequence read is matched there
/home/young/Pictures/2019_06_12_13:20:56.png
  - Meaning
    - That small sequence composed of 7 base pairs comes from that location of reference sequence

- Second small sequence read is matched there
/home/young/Pictures/2019_06_12_13:22:52.png

- /home/young/Pictures/2019_06_12_13:24:22.png

- Depth is 2 (excluding reference sequence)
/home/young/Pictures/2019_06_12_13:24:38.png

- In experiment, depth can be tens of layers, such as 20, 30, and sometimes even over 100

- Suppose "substitution" applied read (substitution variant which someone has, unlike it, generally people have reference-sequence-type order)
- Normal read
/home/young/Pictures/2019_06_12_13:25:59.png
- substitution applied read
/home/young/Pictures/2019_06_12_13:26:53.png

- "substitution variant" is detected
/home/young/Pictures/2019_06_12_13:28:07.png

- See "deletion variant"
- Normal one
/home/young/Pictures/2019_06_12_13:29:08.png
- "Deletion" applied variant
/home/young/Pictures/2019_06_12_13:29:50.png
- This case never be matched into any place of reference sequence
- So, read is splited, to match to reference sequence (which is called split alignment)
/home/young/Pictures/2019_06_12_13:29:50.png
- "deletion" variant is detected
/home/young/Pictures/2019_06_12_13:31:31.png

- See "insertion" variant
- normal
/home/young/Pictures/2019_06_12_13:32:05.png
- "insertion variant" read
/home/young/Pictures/2019_06_12_13:32:39.png
- Split alignment is used 
/home/young/Pictures/2019_06_12_13:33:14.png
/home/young/Pictures/2019_06_12_13:33:39.png

- Insertion variant is detected
/home/young/Pictures/2019_06_12_13:33:51.png

================================================================================
- See another illustration for sequence alignment
/home/young/Pictures/2019_06_12_13:38:31.png

- Chromosom1 as reference sequence
/home/young/Pictures/2019_06_12_13:39:05.png

- One small sequence read
/home/young/Pictures/2019_06_12_13:44:21.png
  - Yellow parts: sequenced part
  - Gray part: not sequenced part
  - When you have "one DNA fragment",
  machine reads it "from front to rear" and "from rear to front"
  - If "one DNA fragment" is too long (such as over 200 bp)
  some parts can't be read (?)
  - So, not-read part is located in gray part (?)

- Refernce sequence has "A"
Fragments have "C"
/home/young/Pictures/2019_06_12_13:47:20.png
  - Meaning:
    - Point mutation (or A to C substitution variant)
    - Remaining parts are same to "reference sequence"

- there are continuous "empty" part
/home/young/Pictures/2019_06_12_13:48:53.png
- This can be machine's error
/home/young/Pictures/2019_06_12_13:49:15.png
- This is "deletion variant" (Indel: insertion and deletion variants)

================================================================================
/home/young/Pictures/2019_06_12_13:51:46.png

Both (homozygous) are deletion variant

Depth 0

================================================================================
/home/young/Pictures/2019_06_12_13:52:28.png

One is normal, one is deletion variant

Depth is 1/2

================================================================================
/home/young/Pictures/2019_06_12_13:53:42.png

Depth is more

Gain (extra copy) variant

================================================================================
/home/young/Pictures/2019_06_12_13:54:16.png

================================================================================
Reads are mapped to both chromosome1 and chromosome5
/home/young/Pictures/2019_06_12_13:54:50.png

Cancer cell often shows this variant

chromosome1 and chromosome5 has fusion due to disease?

fusion point is called "translocation breakpoint"

================================================================================
Some fragments (like bateria or virus) are not mapped to human's reference sequence

/home/young/Pictures/2019_06_12_13:57:29.png

================================================================================
Actual program for sequence alignment

/home/young/Pictures/2019_06_12_13:59:18.png

</xmp>
   </BODY>
</HTML>
