<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
   "http://www.w3.org/TR/html4/strict.dtd">
<HTML>
   <HEAD>
      <TITLE>My first HTML document</TITLE>
      <style rel="stylesheet" type="text/css">
body {
 font-size: 20px;
 
 margin-top: 50px;
    margin-bottom: 50px;
    margin-right: 80px;
    margin-left: 100px;
    
    padding-top: 50px;
    padding-bottom: 50px;
    padding-right: 80px;
    padding-left: 80px;
    
    line-height:35px;
}
/* img {
 width:900px;
} */
</style>
      <script type="text/x-mathjax-config">
MathJax.Hub.Config({
    "HTML-CSS" : {
        availableFonts : ["STIX"],
        preferredFont : "STIX",
        webFont : "STIX-Web",
        imageFont : null
    }
});
</script>
     <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js" type="text/javascript">    
    MathJax.Hub.Config({
        HTML: ["input/TeX","output/HTML-CSS"],
        TeX: { extensions: ["AMSmath.js","AMSsymbols.js"], 
               equationNumbers: { autoNumber: "AMS" } },
        extensions: ["tex2jax.js"],
        jax: ["input/TeX","output/HTML-CSS"],
        tex2jax: { inlineMath: [ ['$$$','$$$'] ],
                   displayMath: [ ['$$$$','$$$$'] ],
                   processEscapes: true },
        "HTML-CSS": { availableFonts: ["TeX"],
                      linebreaks: { automatic: true } }
    });
</script>
   </HEAD>
   <BODY>
<xmp>
https://www.youtube.com/watch?v=TSe-kxFsgto
/mnt/1T-5e7/mycodehtml/Biology/SangSooKim/Bioinformatics_basic/003/002_What_is_exome_Sanger_sequencing/main.html

================================================================================
IRB: institutional review board

IRB checks morality of the research in the each company or school

================================================================================
Exome is captured by using Roche Nimblegen HD2 machine

================================================================================
DNA -> RNA -> (Exons, Introns are removed by splicing algorithm) -> mRNA -> (transcription) -> Protein

================================================================================
Gene: almost 20000 number of genes

One gene has almost 10 exons

20000 * 10 = 200000 exons = Exome

cf) gene1 + gene2 + ... = genome

================================================================================
Exonme capture: collects 200000 exons (with removing all introns)

================================================================================
How to do "exome capture"

/home/young/Pictures/2019_06_12_12:36:13.png

- Onto that array chip, you embed all DNA sequences which you already know 

================================================================================
- Another illustration
/home/young/Pictures/2019_06_12_12:37:31.png

- Embeds all exons sequence (which you already know) onto the chip (gray plate)
/home/young/Pictures/2019_06_12_12:38:03.png

- Randomly cut (patient's) sample DNA
Then, some pieces have exons, but some other pieces will not have exons
/home/young/Pictures/2019_06_12_12:39:10.png

- You insert above "small DNAs" into the gray plate
/home/young/Pictures/2019_06_12_12:41:06.png

- DNA pieces which have exons will attatch to the gray plate
DNA pieces which don't have exons will be swept away
with collecting all exons which correspond to the DNA exon sequences which you already know (or which you're interested in)

================================================================================
Another way which uses beads

- B stands for beads
- You attach "exons" which you already know onto the bead
- Then, you insert "small DNAs" into the bowl in which "beads" are located
- Then, "complement sequence exons" are attached to "existing DNA which is attached to the bead"
/home/young/Pictures/2019_06_12_12:44:29.png

- Then, you perform precipitation for the beads and capture the beads
/home/young/Pictures/2019_06_12_12:49:14.png

- In this way, you can collect exons

================================================================================

</xmp>
   </BODY>
</HTML>
