<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
   "http://www.w3.org/TR/html4/strict.dtd">
<HTML>
   <HEAD>
      <TITLE>My first HTML document</TITLE>
      <style rel="stylesheet" type="text/css">
body {
 font-size: 20px;
 
 margin-top: 50px;
    margin-bottom: 50px;
    margin-right: 80px;
    margin-left: 100px;
    
    padding-top: 50px;
    padding-bottom: 50px;
    padding-right: 80px;
    padding-left: 80px;
    
    line-height:35px;
}
/* img {
 width:900px;
} */
</style>
      <script type="text/x-mathjax-config">
MathJax.Hub.Config({
    "HTML-CSS" : {
        availableFonts : ["STIX"],
        preferredFont : "STIX",
        webFont : "STIX-Web",
        imageFont : null
    }
});
</script>
     <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js" type="text/javascript">    
    MathJax.Hub.Config({
        HTML: ["input/TeX","output/HTML-CSS"],
        TeX: { extensions: ["AMSmath.js","AMSsymbols.js"], 
               equationNumbers: { autoNumber: "AMS" } },
        extensions: ["tex2jax.js"],
        jax: ["input/TeX","output/HTML-CSS"],
        tex2jax: { inlineMath: [ ['$$$','$$$'] ],
                   displayMath: [ ['$$$$','$$$$'] ],
                   processEscapes: true },
        "HTML-CSS": { availableFonts: ["TeX"],
                      linebreaks: { automatic: true } }
    });
</script>
   </HEAD>
   <BODY>
<xmp>

www.bioin.or.kr/fileDown.do?seq=33168
/mnt/1T-5e7/mycodehtml/Biology/Gene_sequencing/Short_tutorial_text/Bioinformatics_driven_analysis_on_next_generation_DNA_sequence_data/main.html

================================================================================
/ 1970ë…„ëŒ€ ìƒì–´(F. Sanger)ì— ì˜í•´ ê°œë°œëœ DNA ì‹œí€€ì‹±(sequencing) ê¸°ìˆ ì€ ì¸ê°„ ìœ ì „ì²´ /
/ í”„ë¡œì íŠ¸(human genome project)ê°€ ì™„ì„±ëœ 2003ë…„ê²½ê¹Œì§€ ë…ë³´ì ìœ¼ë¡œ ì‚¬ìš©ë˜ì—ˆë‹¤. ì´ í›„ 2007ë…„ê²½ /
/ Sanger ì‹œí€€ì‹± ë°©ë²•ì„ ê°œì„ í•œ ì°¨ì„¸ëŒ€ ì—¼ê¸°ì„œì—´ ë¶„ì„ ê¸°ìˆ (next generation sequencing, NGS) [1, 2]ì˜ /
/ ë“±ì¥ìœ¼ë¡œ ì—¼ê¸°ì„œì—´ í•´ë…(sequencing)ì— ê±¸ë¦¬ëŠ” ì‹œê°„ê³¼ ë¹„ìš©ì„ íšê¸°ì ìœ¼ë¡œ ì¤„ì¼ ìˆ˜ ìˆê²Œ ë˜ì—ˆë‹¤. /
/ HTS(high-throughput sequencing) í˜¹ì€ SGS(second-generation sequencing)ë¼ê³ ë„ ë¶ˆë¦¬ëŠ” NGSì˜ /
/ ê°€ì¥ ë‘ë“œëŸ¬ì§„ íŠ¹ì§•ì€ ì´ˆ ë³‘ë ¬(massively parallel)ì ìœ¼ë¡œ ë°±ë§Œì—ì„œ ì‹­ì–µ ë‹¨ìœ„ì˜ ë¦¬ë“œ(reads, ì—¼ê¸°ì„œì—´ /
/ ë‹¨í¸)ë¥¼ ë‹¨ ëª‡ ì‹œê°„ í˜¹ì€ ëª‡ ì¼ ë‚´ë¡œ í•´ë…í•  ìˆ˜ ìˆë‹¤ëŠ” ê²ƒì´ë‹¤.  /

================================================================================
1970-2003: DNA sequencing by F. Sanger

2007-: Next Generation Sequencing


Next_Generation_Sequencing
  - Able to decode millions and 10 billions of reads at the same time during several hours or within several days
  - which is "massively parallel" advantage
  - reads: piece of DNA sequence

================================================================================
/ NGS í”Œë«í¼(platform)ì„ ìƒì‚°í•˜ëŠ” ëŒ€í‘œì ì¸ íšŒì‚¬ë¡œ Roche/454 [3], Illumina/Solexa [4], /
/ ê·¸ë¦¬ê³  Life/APG [5] ë“±ì„ ë“¤ ìˆ˜ ìˆëŠ”ë°, í”Œë«í¼ì€ PCR ì¦í­(amplification) ë°©ì‹ì— ë”°ë¼ í¬ê²Œ ë‘ ì¢…ë¥˜ /
/ (solid-phase amplification, emulsion PCR ë“±)ë¡œ ë‚˜ëˆŒ ìˆ˜ ìˆë‹¤. ë˜í•œ í”Œë«í¼ì€ ë‹¤ë¥´ë”ë¼ë„ ì—¼ê¸°ì„œì—´ì„ /
/ í•´ë…í•˜ëŠ” ì¤‘ìš”í•œ ë‹¨ê³„ì¸ í…œí”Œë¦¿ DNAì˜ ì¤€ë¹„ì™€ ì¦í­, ì´ë¯¸ì§€ ì‘ì—… ë° ì—¼ê¸°ì„œì—´ í•´ë…, ì—¼ê¸°ì„œì—´ /
/ ì •í™•ë„ ê´€ë¦¬(quality control) ë“±ì€ ê³µí†µì ìœ¼ë¡œ ê±°ì¹˜ëŠ” ê³¼ì •ì´ë‹¤. NGSëŠ” ë¹„ë‹¨ DNAë¿ë§Œ ì•„ë‹ˆë¼ RNA, /
/ íŠ¹íˆ RNA ì „ì‚¬ ìˆ˜ì¤€(transcription level)ì„ ì¸¡ì •í•˜ëŠ”ë° ì ìš©í•  ìˆ˜ë„ ìˆë‹¤.  /

================================================================================

Companies which produce "NGS platform"
  - Roche/454, Illumina/Solexa, Life/APG

Category of company based on PCR amplification method
  - solid-phase amplification
  - emulsion PCR

Common procedure
  - Prepare template DNA
  - DNA amplification
  - Image processing
  - Decode DNA sequence
  - Quality control on DNA sequence

Use case by NGS
  - DNA level 
  - RNA level
  - RNA transcription level

================================================================================
/ NGS ë°ì´í„°ëŠ” ë“œ ë…¸ë¸Œ ì–´ì…ˆë¸”ë¦¬(de novo assembly) [8]ì™€ ì°¸ì¡° ì–´ì…ˆë¸”ë¦¬(reference assembly) /
/ í˜¹ì€ ì¬ë¶„ì„(resequencing) ë“± í¬ê²Œ ë‘ ê°€ì§€ ë¶„ì•¼ì— ì ìš©ë  ìˆ˜ ìˆë‹¤. ë‘ ì ìš© ë¶„ì•¼ì˜ ê°€ì¥ í° /
/ ì°¨ì´ì ì€ ì°¸ì¡° ì—¼ê¸°ì„œì—´(reference sequence)ì˜ ìœ ë¬´ì´ë‹¤. ë“œ ë…¸ë¸Œ ì–´ì…ˆë¸”ë¦¬ëŠ” í•´ë‹¹ ì¢…ì˜ ìœ ì „ì²´ /
/ ì—¼ê¸°ì„œì—´ì„ ì²˜ìŒìœ¼ë¡œ í•´ë…í•˜ëŠ” ê²ƒì„ ë§í•˜ë©°, ì°¸ì¡° ì–´ì…ˆë¸”ë¦¬ë€ ì´ë¯¸ ê·œëª…ëœ ì—¼ê¸°ì„œì—´(ì°¸ì¡° /
/ ì—¼ê¸°ì„œì—´)ì— ìƒ˜í”Œì˜ ë¦¬ë“œë¥¼ ì •ë ¬(alignment/mapping)í•˜ëŠ” ê²ƒìœ¼ë¡œ, ìƒ˜í”Œ ê°„ì˜ ìœ ì „ë³€ì´(genetic /
/ variation) ì—°êµ¬ì— ì£¼ë¡œ ì‚¬ìš©ëœë‹¤.  /

================================================================================

NGS_data can be used for:
  - de novo assembly
  - reference assembly (or resequencing)

================================================================================
de novo assembly:
  - Decode DNA sequence on the specific spicies, FOR THE FIRST TIME
  - so, it has no "reference sequence"

reference assembly (or resequencing)
  - it has "reference sequence"
  - performs "alignment/mapping" on reads of sample
  - from "found DNA sequence"
  - it is much used for "genetic variation research" between samples

================================================================================
/ ì¬ë¶„ì„ì„ í†µí•´ ì¶”ì¶œí•  ìˆ˜ ìˆëŠ” ìœ ì „ë³€ì´ ì¤‘ ê°€ì¥ ëŒ€í‘œì ì¸ ê²ƒì´ ë‹¨ì¼ì—¼ê¸°ë‹¤í˜•ì„±(single /
/ nucleotide polymorphism, SNP)ì¸ë°, /

================================================================================
By using resequencing
  - you can extract "genetic variation"
  - for example, single nucleotide polymorphism (SNP)

================================================================================
/ NGSëŠ” ë³´í†µ 100ê°œ ì •ë„ì˜ ì—¼ê¸°ë¡œ êµ¬ì„±ëœ ì§§ì€ ì„œì—´ ì¡°ê°ì¸ ë¦¬ë“œë¥¼ ìƒì„±í•˜ì—¬ ì—¼ê¸°ì„œì—´ì„ /
/ í•´ë…í•œë‹¤. NGSëŠ” í•´ë…í•œ ì—¼ê¸°ì„œì—´ì„ FASTQ í˜•ì‹ [12]ì˜ íŒŒì¼ë¡œ ì €ì¥í•˜ëŠ”ë°, ì´ë¥¼ ë³´í†µ ì›ì‹œ /
/ ë°ì´í„°ë¼ ë¶€ë¥¸ë‹¤. FASTQ íŒŒì¼ì—ì„œ SNP/Indel callingì— ì´ë¥´ëŠ” ê³¼ì •ì€ ë‹¤ì–‘í•œ í†µê³„í•™ ë° ì „ì‚°í•™ì  /
/ ë°©ë²•ì„ ì‚¬ìš©í•˜ëŠ” ì—¬ëŸ¬ ë¶„ì„ ë‹¨ê³„ë¥¼ ê±°ì¹˜ëŠ”ë°, ì´ë¥¼ í”„ë¡œê·¸ë¨í™”í•œ ìƒì—…ìš© í˜¹ì€ ê³µê°œ ë¶„ì„ /
/ íŒŒì´í”„ë¼ì¸ë“¤ì´ ì œê³µë˜ê³  ìˆë‹¤ [13]. ì›ì‹œ ë°ì´í„°ì—ì„œ SNP/Indel callingì— ì´ë¥´ëŠ” ê³¼ì •ì€ í¬ê²Œ 3ë‹¨ê³„, /
/ ì¦‰ ì›ì‹œ ë°ì´í„° ìƒì„±(raw data generation), ì°¸ì¡° ì—¼ê¸°ì„œì—´ì— ë¦¬ë“œì˜ ì •ë ¬(alignment/mapping), /
/ ê·¸ë¦¬ê³  ìœ ì „ë³€ì´ ì¶”ì¶œ(variant calling) ë“±ìœ¼ë¡œ êµ¬ë¶„í•  ìˆ˜ ìˆê³  [14, 15], ê° ë‹¨ê³„ì˜ ê²°ê³¼ë¬¼ë¡œ ì„¸ê°€ì§€ /
/ í˜•ì‹(FASTQ, SAM/BAM, VCF)ì˜ íŒŒì¼ì´ ìˆë‹¤. / 

================================================================================
In NGS:
  # ================================================================================
  # @ Step1
  reads=create_reads(DNA)

  num_bases=how_many_bases(reads)
  # num_bases: around 100 bases

  decoded_DNA_sequence=NGS_sequencing(num_bases)

  fastq_file_as_raw_data=save_as_FASTQ_format(decoded_DNA_sequence)

  # ================================================================================
  # @ Step2

  SAM_BAM_format_file=alignment_and_mapping(reads_of_reference_sequence)

  # ================================================================================
  # @ Step3

  VCF_format_file=extract_genetic_variant(fastq_file_as_raw_data,SAM_BAM_format_file)

================================================================================
/ë˜í•œ ì›ì‹œ ë°ì´í„° ìƒì„±ê³¼ ë¦¬ë“œì˜ ì •ë ¬ì€ ê° ë¦¬ë“œì˜ /
/ ì—¼ê¸°ì„œì—´ ì •ë³´ë¥¼ ì²˜ë¦¬í•˜ëŠ” ë‹¨ê³„ [ê·¸ë¦¼ 1ì˜ ì™¼ìª½]ì— í•´ë‹¹í•˜ë©°, ìœ ì „ë³€ì´ ì¶”ì¶œì€ ëª¨ë“  ë¦¬ë“œì— ëŒ€í•œ /
/ ì •ë³´ ì²˜ë¦¬ ê²°ê³¼ë¥¼ í†µí•©í•˜ëŠ” ë‹¨ê³„ [ê·¸ë¦¼ 1ì˜ ì˜¤ë¥¸ìª½]ì— í•´ë‹¹í•œë‹¤. /

================================================================================
Overall pipeline for SNP/Indel calling
/home/young/Pictures/2019_06_07_08:47:08.png

- Perform following process on "raw data", "reference sequence"
/home/young/Pictures/2019_06_07_08:48:16.png

- Extract genetic variation by comparing data
/home/young/Pictures/2019_06_07_08:49:19.png

================================================================================
/ ëŒ€ë¶€ë¶„ì˜ íŒŒì´í”„ë¼ì¸ì€ ìœ„ì˜ 3ë‹¨ê³„ë¥¼ ì¢€ ë” ì„¸ë°€í•˜ê²Œ êµ¬ë¶„(base calling and initial QC, /
/ alignment to reference, post-processing of alignment, quality score recalibration, SNP calling, filtering /
/ of SNP candidate ë“±)í•˜ì—¬ ë°ì´í„° ë¶„ì„ì„ ì§„í–‰í•œë‹¤ [13, 16, 17]. /

Detail steps of 3rd following "data analysis" procedure
/home/young/Pictures/2019_06_07_08:51:44.png

- base calling and initial QC
- alignment to reference
- post-processing of alignment
- quality score recalibration
- SNP calling
- filtering of SNP candidate

================================================================================
/ Broad Instituteì—ì„œëŠ” /
/ GATK(Genome Analysis Toolkit) [18]ë¼ ë¶ˆë¦¬ëŠ” ì†Œìœ„ â€˜ëª¨ë²”ì‹¤ë¬´â€™(best practice) SNP/Indel calling /
/ íŒŒì´í”„ë¼ì¸ í™˜ê²½ì„ ê³µê°œí•˜ê³  ìˆëŠ”ë°, GATKëŠ” í˜„ì¬ NGS ë°ì´í„° ë¶„ì„ì— ê°€ì¥ ë§ì´ ì‚¬ìš©ë˜ê³  ìˆëŠ” /
/ íŒŒì´í”„ë¼ì¸ì´ë‹¤. /

Best practice pipeline for SNP/Indel calling:
  - Genome Analysis Toolkit (GATK) of Broad Institute

================================================================================
/ NGS ë¦¬ë“œì˜ ê¸¸ì´ëŠ” ì•½ 100 bp ì •ë„ë¡œ ê¸°ì¡´ Sanger íƒ€ì…ì˜ 500-1,000 bpì— ë¹„í•˜ì—¬ ê¸¸ì´ê°€ /
/ ì§§ê³ , ì‹œí€€ì‹± ì˜¤ë¥˜ê°€ ìƒëŒ€ì ìœ¼ë¡œ í¬ë©°, í”Œë«í¼ì— ì˜ì¡´í•˜ëŠ” ì˜¤ë¥˜ë„ í¬í•¨ë  ìˆ˜ ìˆë‹¤. /

================================================================================
NGS:
  - length of read: around 100 bp
  - length of read from Sanger sequencing: 500-1000 bp
  - large sequencing error rate
  - contains platform-driven-errors

================================================================================
/NGS í”Œë«í¼ë“¤ì´ /
/ ìƒì„±í•˜ëŠ” FASTQ íŒŒì¼ì€ ê¸°ì¡´ì˜ DNA ì—¼ê¸°ì„œì—´ì„ ë‚˜íƒ€ë‚´ëŠ” í…ìŠ¤íŠ¸ ê¸°ë°˜ì˜ í‘œì¤€ ì—¼ê¸° ë°ì´í„° í˜•ì‹ì¸ /
/ FASTA í˜•ì‹ [19]ì— í•´ë…í•œ ì—¼ê¸°ì˜ ì •í™•ë„(quality score í˜¹ì€ error rate)ë¥¼ í¬í•¨ì‹œí‚¨ ê²ƒì´ë‹¤. /

================================================================================
FASTQ file from NGS
  - FASTA data (DNA sequence)
  - quality scroe (or error rate)

================================================================================
/ê° ë¦¬ë“œ /
/ ë‹¹ ìƒì„±ë˜ëŠ” FASTQ íŒŒì¼ì€ 4 ì¤„ë¡œ êµ¬ì„±ë˜ëŠ”ë°, ì²«ì§¸ ì¤„ì€ @ìœ¼ë¡œ ì‹œì‘í•˜ë©° ì‚¬ìš©í•œ í”Œë«í¼ê³¼ /
/ ì—¼ê¸°ì„œì—´ ê¸¸ì´ ë“±ì— ëŒ€í•œ ì •ë³´ë¥¼ í¬í•¨í•˜ê³  ìˆê³ , ë‘˜ì§¸ ì¤„ì€ í•´ë…í•œ ì—¼ê¸°ì„œì—´, ì…‹ì§¸ ì¤„ì€ + ê¸°í˜¸ë¡œ /
/ ì‹œì‘í•˜ë©° ê¸°íƒ€ ì„¤ëª…, ê·¸ë¦¬ê³  ë§ˆì§€ë§‰ ì¤„ì€ ë‘˜ì§¸ ì¤„ì˜ ì—¼ê¸°ì„œì—´ì— ëŒ€í•œ ì •í™•ë„(quality score)ë¥¼ /
/ í‘œì‹œí•œë‹¤ [ê·¸ë¦¼ 2]. ë”°ë¼ì„œ ë‘˜ì§¸ ì¤„ê³¼ ë„·ì§¸ ì¤„ì€ ê°™ì€ ê°œìˆ˜ ì •ë³´ë¡œ êµ¬ì„±ëœë‹¤. /

================================================================================
/home/young/Pictures/2019_06_07_10:26:40.png

- FASTQ file for "each read" is composed of 4 sentences
/home/young/Pictures/2019_06_07_10:27:56.png

- First row (title identifier description) starts with @
/home/young/Pictures/2019_06_07_10:28:36.png

- First row (title identifier description):
  - used platform
  - length of sequence
  - 1: paired-end: forward (in Illumina case)
  - 2: paired-end: reverse (in Illumina case)

- Second row is "decoded sequence"
/home/young/Pictures/2019_06_07_10:31:07.png

- Third row (as + description) starts with +
/home/young/Pictures/2019_06_07_10:33:02.png

- + description explains miscellaneous information

- Forth row is quality score value with respect to "decoded sequence in second row"
/home/young/Pictures/2019_06_07_10:35:01.png

- Length of second row = length of forth row

================================================================================
/ ë˜í•œ Illumina í”Œë«í¼ì€ /
/ ë¦¬ë“œì˜ ì–‘ìª½ ëì—ì„œ ê°ê° ì—¼ê¸°ì„œì—´ì„ í•´ë…í•˜ì—¬ ë¦¬ë“œ ë‹¹ ê°™ì€ í¬ê¸°ì˜ FASTQ íŒŒì¼ í•œ ìŒ(forward and /
/ reverse)ì„ ìƒì„±í•œë‹¤. ì´ê²ƒì„ paired-end readsë¼ ë¶€ë¥¸ë‹¤ [ê·¸ë¦¼ 2]. /

================================================================================
Illumina platform:
  - Decode sequence in both ends of the reads
  - FASTQ_file_1_for_forward_end
  - FASTQ_file_2_for_reverse_end
  - paired-end reads

================================================================================
- FASTQ_file_1_for_forward_end
/home/young/Pictures/2019_06_07_10:37:06.png

- FASTQ_file_2_for_reverse_end
/home/young/Pictures/2019_06_07_10:37:26.png

================================================================================
/ FASTQ íŒŒì¼ì´ í¬í•¨í•˜ê³  ìˆëŠ” ì—¼ê¸°ì„œì—´ì€ SNP/Indel callingì„ ìœ„í•œ í›„ì† ë¶„ì„ ê³¼ì •ë“¤ì— /
/ ì§€ì†ì ìœ¼ë¡œ ì˜í–¥ì„ ë¯¸ì¹˜ê¸° ë•Œë¬¸ì— ì—¼ê¸°ì„œì—´ì˜ ì •í™•ë„ëŠ” ë§¤ìš° ì¤‘ìš”í•˜ë‹¤.  /

Sequence in FASTQ file
  - continuously affects postprocessing for SNP/Indel calling
  - accuracy of sequence in FASTQ file is important

================================================================================
/ ë˜í•œ SNPëŠ” ì¸ê°„ì˜ ê²½ìš° /
/ ì „ì²´ ì§€ë†ˆ(genome)ì˜ ì•½ 0.1% (ì•½ 1,000bp ì¤‘ 1ê°œ) ì •ë„ ë°–ì— ë‚˜íƒ€ë‚˜ì§€ ì•Šìœ¼ë¯€ë¡œ ì´ë¥¼ í™•ì¸í•˜ëŠ” /
/ ê¸°ìˆ ì€ ëŒ€ë‹¨íˆ ì •í™•í•´ì•¼ í•˜ë©°, ì‹œí€€ì‹± ì˜¤ë¥˜ëŠ” ë¶€ì •í™•í•œ SNP/Indel callingìœ¼ë¡œ ì´ì–´ì§ˆ ìˆ˜ ìˆë‹¤. /

================================================================================
SNP of human:
  - 0.1% out of entire genome
  - "1 bp out 1000bp" ratio
  - so, detecting SNP should be highly accurate
  - incorrect sequencing DNA -> incorrect detecting SNP -> incorrect SNP/Indel calling

================================================================================
/ ì¼ë°˜ì ìœ¼ë¡œ ë¦¬ë“œ ì—¼ê¸°ì„œì—´ì˜ ê¸¸ì´ê°€ ì»¤ì§ì— ë”°ë¼ í•´ë…í•œ ì—¼ê¸°ì˜ ì˜¤ë¥˜ìœ¨ì´ ì¦ê°€í•¨ì´ ì•Œë ¤ì ¸ ìˆë‹¤ [9]. /

Length of the read increases
-> decoded sequence's error rate increases

================================================================================
ê·¸ë¦¼ 3ì—ì„œ ë¶ˆ ìˆ˜ ìˆë“¯ì´ ë¦¬ë“œì˜ ì—¼ê¸° ìœ„ì¹˜(í˜¹ì€ sequence cycle)ê°€ ì»¤ì§ì— ë”°ë¼ í‰ê·  ì˜¤ë¥˜ìœ¨ì€
ì¦ê°€í•˜ë©°, íŠ¹íˆ í‘œì¤€í¸ì°¨ëŠ” í‰ê·  ì˜¤ë¥˜ìœ¨ ë³´ë‹¤ ë” ê°€íŒŒë¥´ê²Œ ì¦ê°€í•œë‹¤. ì´ê²ƒì€ sequence cycleì´
ì¦ê°€í•¨ì— ë”°ë¼ ì˜¤ë¥˜ìœ¨ì˜ ë³€ë™ì´ ì‹¬í•˜ê³ , í›„ì† ë¶„ì„ì„ ìœ„í•œ ë°ì´í„°ë¡œ ì‚¬ìš©í•˜ê¸° ì–´ë ¤ìš´ ë§¤ìš° í°
ì˜¤ë¥˜ì˜ ì—¼ê¸°ë„ í¬í•¨ë¨ì„ ì˜ë¯¸í•œë‹¤.

/home/young/Pictures/2019_06_07_10:49:00.png

================================================================================
/ ëŒ€ë¶€ë¶„ì˜ ë¶„ì„ íŒŒì´í”„ë¼ì¸ì€ í›„ì† ë¶„ì„ì˜ ì •í™•ë„ë¥¼ ë†’ì´ê¸° ìœ„í•´ NGS ì—¼ê¸°ì„œì—´ì˜ ì •í™•ë„ë¥¼ /
/ ê´€ë¦¬í•œë‹¤. /

For high accuracy in postprocessing in sequecing pipeline,
people manage accuracy of "NGS sequence"

================================================================================
/ ì—¼ê¸°ì„œì—´ì˜ ì •í™•ë„ ê´€ë¦¬ëŠ” ì–´ëŒ‘í„°(adaptor) ì„œì—´ì„ ì œê±°í•˜ê±°ë‚˜ ì •í™•ë„ê°€ í˜„ì €íˆ ë‚®ì€ ì—¼ê¸° /
/ (ì˜ˆë¥¼ ë“¤ì–´, ğ‘„ğ‘ƒâ„ğ‘Ÿğ‘Ÿğ‘Ÿ < 10 ì¸ ì—¼ê¸°)ë¥¼ ì œê±°í•˜ëŠ” ê²ƒì„ í¬í•¨í•œë‹¤. /

Management of accuracy of decoded sequence 
  - remove adaptor
  - remove sequence with low accuracy like $$$Q_{Phred} \lt 10$$$

================================================================================
/ Illumina í”Œë«í¼ì˜ ê²½ìš°ì—ëŠ” í”Œë«í¼ /
/ ìì²´ì ìœ¼ë¡œ ë¶€ë¶„ì ì´ë‚˜ë§ˆ ì •í™•ë„ ê´€ë¦¬ë¥¼ í•˜ê³ , SOLiD í”Œë«í¼ì˜ ê²½ìš°ì—ëŠ” ì¶”í›„ ì •ë ¬ ê³¼ì •ì—ì„œ /
/ ì •í™•ë„ê°€ ë‚®ì€ ì—¼ê¸°ëŠ” ê±¸ëŸ¬ì§ˆ ìˆ˜ ìˆê¸° ë•Œë¬¸ì— íŠ¹ë³„í•œ ì •í™•ë„ ê´€ë¦¬ë¥¼ í•˜ì§€ ì•ŠëŠ” ê²ƒìœ¼ë¡œ ì•Œë ¤ì ¸ ìˆë‹¤ /
/ [16]. /

Illumina platform
  - include accuracy-management-pipeline in their entire pipeline

SOLiD platform
  - in alignment/mapping step,
  - sequence which has low accuracy is removed

================================================================================
/ í˜„ì¬ ì—¼ê¸°ì˜ ì •í™•ë„ ê´€ë¦¬ë¥¼ ìœ„í•œ ë‹¤ì–‘í•œ ì†Œí”„íŠ¸ì›¨ì–´ê°€ ê°œë°œë˜ì–´ ì‚¬ìš©ë˜ê³  ìˆëŠ”ë° [í‘œ 1], /
/ ìë°”(JAVA) ì–¸ì–´ë¡œ êµ¬í˜„ëœ FastQCëŠ” í˜„ì¬ ê°€ì¥ ë„ë¦¬ ì‚¬ìš©ë˜ëŠ” ì†Œí”„íŠ¸ì›¨ì–´ì´ë©°, qrqcì€ R ì–¸ì–´ë¡œ /
/ êµ¬í˜„í•œ íŒ¨í‚¤ì§€ì´ë‹¤. /

================================================================================
Software for accuracy-management

- FastQC
  - most used
  - Java
- qrqc
  - R

/home/young/Pictures/2019_06_07_10:56:28.png

================================================================================
/ íŠ¹íˆ ì •í™•ë„ê°€ ë‚®ì€ ì—¼ê¸°ë¥¼ FASTQ íŒŒì¼ì—ì„œ ì œê±°í•˜ëŠ” ì‘ì—…ì„ /
/ ê°€ì§€ì¹˜ê¸°(trimming)ë¼ í•˜ëŠ”ë°, ê°€ì§€ì¹˜ê¸°ëŠ” ê¸°ì¤€ ì—¼ê¸°ì„œì—´ì— ì •ë ¬í•˜ê¸° ì „ì— ì‹¤ì‹œí•  ìˆ˜ë„ ìˆìœ¼ë©°, BWA /
/ ì•Œê³ ë¦¬ì¦˜ê³¼ ê°™ì´ ì •ë ¬ ê³¼ì •ì—ì„œ ê°€ì§€ì¹˜ê¸°ë¥¼ í•  ìˆ˜ë„ ìˆë‹¤. /

Trimming:
  - remove sequences (which have low accuracy) from FASTQ file
  - Scenario1
  Trimming -> Alignment/mapping
  - Scenario2
  Trimming & Alignment/mapping at the same time

================================================================================
/ ì •ë ¬(alignment/mapping)ì€ ìœ ì „ë³€ì´ì™€ ì‹œí€€ì‹± ì˜¤ë¥˜ë¥¼ í¬í•¨í•˜ê³  ìˆëŠ” 106~109 ê°œì˜ /
/ ë¦¬ë“œë“¤ì„ ì°¸ì¡° ì—¼ê¸°ì„œì—´ê³¼ ë¹„êµí•˜ì—¬ ë¦¬ë“œì˜ ì—¼ê¸°ì„œì—´ê³¼ ì¼ì¹˜í•˜ëŠ” ìœ„ì¹˜ë¥¼ ì°¸ì¡° ì—¼ê¸°ì„œì—´ì—ì„œ ì°¾ëŠ” /
/ ê³¼ì •ì´ë‹¤. /

================================================================================
alignment/mapping
  - you have 106-109 number of reads
  - which contains "genetic variation" and "sequence error"

  - you compare them with "reference sequence"
  - you find the locations where "reads" and "reference sequence" are identical

================================================================================
/ ì •ë ¬ì—ì„œ ê°€ì¥ í° ë¬¸ì œëŠ” ë¦¬ë“œì˜ ê¸¸ì´ê°€ ì§§ê³  ì°¸ì¡° ì—¼ê¸°ì„œì—´ ìì²´ì— ë°˜ë³µ ì˜ì—­(repetitive /
/ regions)ì´ ìˆìŒìœ¼ë¡œ ì°¸ì¡° ì—¼ê¸°ì„œì—´ì˜ ì—¬ëŸ¬ ê³³ì— ë™ì‹œì— ì •ë ¬ë  ìˆ˜ ìˆëŠ” ê°€ëŠ¥ì„±ì´ë‹¤ /

================================================================================
Most dangerous issue in alignment
  - reads have short length
  - reference sequence has long length
  - which has "repetitive regions"
  - therefore, "short reads" can be aligned to multiple regions of "reference sequence"


================================================================================
/ ë‘ ì—¼ê¸°ì„œì—´ì„ /
/ ë¹„êµí•˜ëŠ”ë° ì¼ë°˜ì ìœ¼ë¡œ ë™ì  í”„ë¡œê·¸ë˜ë°(dynamic programming) ì•Œê³ ë¦¬ì¦˜ì— ê¸°ì´ˆí•œ BLAST(basic /
/ local alignment search tool)ì™€ ê°™ì€ ì†Œí”„íŠ¸ì›¨ì–´ë¥¼ ì‚¬ìš©í•  ìˆ˜ ìˆìœ¼ë‚˜, BLASTëŠ” ê³„ì‚°ëŸ‰ì´ ë„ˆë¬´ ë§ì•„ì„œ /
/ NGS ë°ì´í„°ì˜ ì •ë ¬ì—ëŠ” ì‚¬ìš©í•˜ì§€ ì•ŠëŠ”ë‹¤. /

================================================================================
To compare "reads" and "reference sequence"
  - you can use "BLAST (basic local alignment search tool)" 
  - which is based on "dynamic programming" methodology
  - but BLAST has too much computation
  - so BLAST is not used to align "NGS data"

================================================================================
/ NGS ë°ì´í„°ì˜ ì •ë ¬ì„ ìœ„í•œ ì•Œê³ ë¦¬ì¦˜ì€ í¬ê²Œ /
/ BWT(Burrows-Wheeler transformation)ì™€ í•´ì‹±(hashing)ì— ê¸°ë°˜í•œ ì•Œê³ ë¦¬ì¦˜ ë“± ë‘ ê°€ì§€ ìœ í˜•ìœ¼ë¡œ /
/ êµ¬ë¶„í•  ìˆ˜ ìˆëŠ”ë°, ë‘ ì•Œê³ ë¦¬ì¦˜ ëª¨ë‘ BLASTì— ë¹„í•´ ê³„ì‚° ì†ë„ê°€ 1,000-10,000ë°° ì •ë„ë¡œ ë¹ ë¥¸ /
/ ì•Œê³ ë¦¬ì¦˜ì´ë‹¤. /

================================================================================
2 algorithms to align "NGS data"

- BWT (Burrows-Wheeler transformation)
- Algorithm which is based on hashing
- Both are 1000-10000x faster than BLAST

================================================================================
/ BWTëŠ” ì›ë˜ ë°ì´í„° ì••ì¶•ì— ì‚¬ìš©ë˜ëŠ” ë³€í™˜ìœ¼ë¡œ ì°¸ì¡° ì—¼ê¸°ì„œì—´(ì „ì²´ ë¬¸ìì—´ì— í•´ë‹¹í•¨)ì„ /
/ ì ‘ë¯¸ì‚¬(suffix)ì˜ íŠ¹ë³„í•œ ìˆœì—´(permutation) í˜•íƒœë¡œ ë³€í™˜í•˜ê³ , ë³€í™˜ëœ í˜•íƒœì—ëŠ” ë™ì¼í•œ ì—¼ê¸°ê°€ /
/ ì—°ì†ì ìœ¼ë¡œ ë‚˜íƒ€ë‚˜ëŠ” íŠ¹ì§•ì„ ì´ìš©í•˜ì—¬ ë¦¬ë“œ ì—¼ê¸°ì„œì—´(ë¶€ë¶„ ë¬¸ìì—´ì— í•´ë‹¹)ì´ í¬í•¨ëœ êµ¬ê°„ì„ /
/ ê²€ìƒ‰í•œë‹¤.  /

================================================================================
BWT is tranformation to compress the data

- BWT converts "reference sequence" into specific permutation form based on suffix
- Same sequence parts show repretively
- Index the parts in the "reference sequence" to which "reads" is identical

================================================================================
/ ì´ì— ë°˜í•˜ì—¬ í•´ì‹± ë°©ë²•ì€ ì°¸ì¡° ì—¼ê¸°ì„œì—´ê³¼ ë¦¬ë“œ ì—¼ê¸°ì„œì—´ì„ ë” ì§§ì€ ê¸¸ì´ë¡œ ìë¥¸ í›„ í•´ì‹œ /
/ í•¨ìˆ˜(hash function)ë¥¼ ì´ìš©í•˜ì—¬ í•´ì‹œ í…Œì´ë¸”ì˜ ê°’ìœ¼ë¡œ í‘œí˜„í•˜ëŠ” ê²ƒìœ¼ë¡œ, ë‘ ì—¼ê¸°ì„œì—´ì˜ ìœ ì‚¬ì„±ì´ /
/ í•´ì‹œ í…Œì´ë¸”ì˜ ê°™ì€ ê°’ìœ¼ë¡œ í‘œí˜„ë¨ì„ ì´ìš©í•œ ê²ƒì´ë‹¤.  /

- Cut "reference sequence" and "reads" into more shorter lengths
- Use "hash function" to express cut "reference sequence" and "reads" as hash table value
- If 2 sequences from "reference sequence" and "reads" are similar,
- they're represented by same valu in hash table

================================================================================
/ BWTë¥¼ ì‚¬ìš©í•˜ëŠ” ì†Œí”„íŠ¸ì›¨ì–´ë¡œ Bowtie, Bowtie2 /
/ [21], BWA [22] ë“±ì´ ìˆìœ¼ë©°, í•´ì‹± ë°©ë²•ì„ ì‚¬ìš©í•˜ëŠ” ì†Œí”„íŠ¸ì›¨ì–´ë¡œ MAQ [23], Stampy [24], SOAP [25] /
/ ë“±ì´ ìˆë‹¤ [í‘œ 2]. /

Software using BWT
  - Bowtie
  - Bowtie2
  - BWA

Software using hasing  
  - MAQ
  - Stampy

================================================================================
/home/young/Pictures/2019_06_07_11:39:43.png

================================================================================
/ ì •ë ¬ ì•Œê³ ë¦¬ì¦˜ì˜ ì„±ëŠ¥ì€ ì²˜ë¦¬ëŸ‰(throughput)ê³¼ ì •í™•ë„(accuracy)ë¡œ ë‚˜íƒ€ë‚¼ ìˆ˜ ìˆë‹¤. ì²˜ë¦¬ëŸ‰ì€ /
/ ì •ë ¬ ì†ë„ë¡œ ê°™ì€ ì‹œê°„ ë‚´ì— ë³´ë‹¤ ë§ì€ ë¦¬ë“œë¥¼ ì •ë ¬í•˜ëŠ” ëŠ¥ë ¥ì„ ë§í•˜ë©°, ì •í™•ë„ëŠ” ì‹œí€€ì‹± /
/ ê³¼ì •ì—ì„œì˜ ì˜¤ë¥˜ì™€ ì‹¤ì œ ì—¼ê¸°ì„œì—´ì˜ ë³€ì´ì— ì ì ˆíˆ ëŒ€ì‘í•˜ë©´ì„œ ë¦¬ë“œì˜ ì •í™•í•œ ì •ë ¬ ìœ„ì¹˜ë¥¼ /
/ ì°¾ì•„ë‚´ëŠ” ëŠ¥ë ¥ì„ ê°€ë¦¬í‚¨ë‹¤.  /

Performance of "alignment algorithm"
- throughput
  - amount of processing reads

- accuracy
  - properly responds to actual genetic variant of reads, 
  by comparing it to the "reference sequence"
  - at the same time, it should properly respond to the error 
  - with distinguishing above "actual genetic variant" and "error"
  - and finally, high accuracy should find the "alignment place" in reference sequence
  - to which "reads" shound be aligned to 

================================================================================
/ ì²˜ë¦¬ëŸ‰ ì¸¡ë©´ì—ì„œ ë³´ë©´ BWTì— ê¸°ë°˜í•œ ì •ë ¬ ì•Œê³ ë¦¬ì¦˜ì´ í•´ì‰¬ í…Œì´ë¸”ì„ /
/ ì‚¬ìš©í•œ ì•Œê³ ë¦¬ì¦˜ ë³´ë‹¤ ì•½ 10ë°° ì •ë„ ë¹ ë¥´ë‚˜, ì •í™•ë„ ì¸¡ë©´ì—ì„œëŠ” í•´ì‹± ì•Œê³ ë¦¬ì¦˜ì´ ìš°ìˆ˜í•˜ë‹¤. /
/ ì¦‰, BWT /
/ ê¸°ë°˜ ì•Œê³ ë¦¬ì¦˜ì€ ë³€ì´ê°€ ì ì€ ë°ì´í„°ì—ì„œëŠ” ë™ë“±í•œ ìˆ˜ì¤€ì˜ ì •í™•ë„ë¥¼ ê°€ì§€ë©´ì„œë„ ë” ë¹ ë¥¸ ì²˜ë¦¬ /
/ ì†ë„ë¥¼ ë³´ì´ëŠ” ë°˜ë©´, í•´ì‹± ê¸°ë°˜ ì•Œê³ ë¦¬ì¦˜ì€ ì—¼ê¸° ë³€ì´ê°€ ë§ì€ ë°ì´í„°ì—ì„œ ë” ë§ì€ ë¦¬ë“œë¥¼ /
/ ì„±ê³µì ìœ¼ë¡œ ì •ë ¬í•œë‹¤ [24] /

- Amount of aligning "reads"
BWT is 10x faster tha hasing

- Accuracy is better in hashing than BWT

- Good case for BWT
  - target data which has low number of genetic variants

- Good case for hashing
  - target data which has many number of genetic variants

================================================================================
/ íŠ¹íˆ paired-end /
/ readì˜ ê²½ìš°ì—ëŠ” forward ë¦¬ë“œì™€ reverse ë¦¬ë“œ ê°ê°ì„ ë…ë¦½ì ìœ¼ë¡œ ì •ë ¬í•˜ê³  ê° ë¦¬ë“œê°€ ì •ë ¬ëœ /
/ ìœ„ì¹˜ì— ëŒ€í•œ ì •ë³´(.sai íŒŒì¼)ë¥¼ êµ¬í•œ í›„, ì´ë¥¼ ì´ìš©í•˜ì—¬ ë‘ ë¦¬ë“œì˜ ì •ë ¬ ê²°ê³¼ë¥¼ í•©ì¹œë‹¤. /

================================================================================
paired-end read case

- align forward read, create .sai file which has information of alignment of read
- align reverse read, create .sai file which has information of alignment of read
- merge 2 alignment results

================================================================================
/ ì •ë ¬ ì•Œê³ ë¦¬ì¦˜ ì„ íƒë³´ë‹¤ ë” ì¤‘ìš”í•œ ê²ƒì€ ë§¤ê°œë³€ìˆ˜(parameter) ê°’ì˜ ê²°ì •ì´ë‹¤. /

================================================================================
Your choice
  - alignment algorithm
  - parameter values in selected alignment algorithm

================================================================================
/ ì¤‘ìš”í•œ /
/ ë§¤ê°œë³€ìˆ˜ë¡œ ì°¸ì¡° ì—¼ê¸°ì„œì—´ê³¼ ë¦¬ë“œ ì‚¬ì´ì˜ ë¶ˆì¼ì¹˜(mismatch) ì—¼ê¸° ê°œìˆ˜ë¥¼ ë“¤ ìˆ˜ ìˆë‹¤. ë¶ˆì¼ì¹˜ë¥¼ /
/ í—ˆìš©í•˜ì§€ ì•Šê³  ì™„ë²½í•œ ì¼ì¹˜ë§Œ í—ˆìš©í•œë‹¤ë©´ í›„ì† ë¶„ì„ì—ì„œëŠ” SNP/Indelì€ ê²€ì¶œí•  ìˆ˜ ì—†ê²Œ ëœë‹¤. ì´ì™€ /
/ ë°˜ëŒ€ë¡œ ë„ˆë¬´ í° ë¶ˆì¼ì¹˜ë¥¼ í—ˆìš©í•˜ë©´ ì˜ëª»ëœ ì •ë ¬ ê²°ê³¼ë¥¼ ë‚˜ì„ ìˆ˜ ìˆê³ , ê·¸ ê²°ê³¼ë¡œ ë§ì€ /
/ ê±°ì§“ì–‘ì„±(false-positive) SNP/Indelì„ ì–»ê²Œ ëœë‹¤. ë”°ë¼ì„œ ì ì ˆí•œ ë¶ˆì¼ì¹˜ ì—¼ê¸° ê°œìˆ˜ë¥¼ ê²°ì •í•˜ëŠ” ê²ƒì´ /
/ ì¤‘ìš”í•œ ë¬¸ì œë¡œ ë¶€ê°ëœë‹¤. /

Mismatch parameter:
  - mismatching threshold value between "reference sequence" and "read"
  - too low mismatching threshold value (perfect matching)
  - in postprocessing, you can't find "SNP/Indel"
  - too high mismatching threshold value (losen matching criterion)
  - can create incorrect alignment result (many false-positive SNP/Indel result)

================================================================================
ìœ ì „ì²´ëŠ” ë¦¬ë“œì˜ ê¸¸ì´ì™€ ìœ ì‚¬í•˜ê±°ë‚˜ ë” ê¸´ ê¸¸ì´ì˜ ë°˜ë³µ ì—¼ê¸°ì„œì—´(repetitive
sequence)ì´ ì¡´ì¬í•˜ê¸° ë•Œë¬¸ì—, í•˜ë‚˜ì˜ ë¦¬ë“œê°€ ì°¸ì¡° ì—¼ê¸°ì„œì—´ì˜ ì—¬ëŸ¬ ìœ„ì¹˜ì— ë™ì¼í•œ ì •í™•ë„ë¡œ ì •ë ¬ë 
ìˆ˜ ìˆë‹¤. ë˜í•œ ëª‡ ê°œì˜ ëŒì—°ë³€ì´(mutation) í˜¹ì€ ì˜¤ë¥˜ë¡œ ì¸í•˜ì—¬ ë‹¤ë¥¸ ìœ„ì¹˜ì— ì •ë ¬ë  ìˆ˜ ìˆë‹¤. ë”°ë¼ì„œ
ì •ë ¬ ê³¼ì •ì´ ëë‚œ ë‹¤ìŒ ì •ë ¬ ê²°ê³¼ë¥¼ ë‹¤ì–‘í•œ í†µê³„ì¹˜ë¥¼ ì‚¬ìš©í•˜ì—¬ ê²€í† í•˜ëŠ” ê²ƒì´ í•„ìš”í•˜ë‹¤. ì˜ˆë¥¼ ë“¤ë©´
ê¸°ì¤€ ì—¼ê¸°ì„œì—´ì— ì •ë ¬ëœ ë¦¬ë“œì˜ ë¹„ìœ¨, paired-end ë¦¬ë“œì˜ ê²½ìš°ì—ëŠ” ì„±ê³µì ìœ¼ë¡œ ìŒ(pair)ì„ ì´ë£¬ ë¹„ìœ¨,
ê·¸ë¦¬ê³  paired-endì˜ ì‚½ì… í¬ê¸°(insert size)ì— ëŒ€í•œ ë¹ˆë„ ë“±ì„ ë“¤ ìˆ˜ ìˆë‹¤. ì´ëŸ¬í•œ í†µê³„ì¹˜ëŠ” ì •ë ¬ì„
ìœ„í•œ ë§¤ê°œë³€ìˆ˜ ê²°ì •ì— ë„ì›€ì„ ì¤„ ìˆ˜ ìˆìœ¼ë©°, ì •í™•ë„ë¥¼ ë†’ì´ëŠ”ë° ê¸°ì—¬í•  ìˆ˜ ìˆë‹¤.

================================================================================
/ ë¦¬ë“œë“¤ì˜ ì •ë ¬ ê²°ê³¼ëŠ” SAM(sequence alignment map) í˜•ì‹ [26]ìœ¼ë¡œ ì €ì¥ëœë‹¤. /

Alignment result of "reads" is saved into SAM (sequence alignment map) file

================================================================================
/ SAM íŒŒì¼ì€ /
/ ìµœì†Œ 11ê°œì˜ í•„ë“œëŠ” ë°˜ë“œì‹œ ê°€ì§€ê³  ìˆëŠ”ë°, ë¦¬ë“œì´ë¦„, ë¦¬ë“œê°€ ì°¸ì¡° ì—¼ê¸°ì„œì—´ì— ì •ë ¬ëœ ê²°ê³¼ë¥¼ /
/ ë‚˜íƒ€ë‚´ëŠ” bitwise flags, ì°¸ì¡° ì—¼ê¸°ì„œì—´ ì´ë¦„, ì •ë ¬ëœ ìœ„ì¹˜, ì •ë ¬ ì •í™•ë„(mapping quality), CIGAR ë“±ì˜ /
/ ì •ë³´ë¥¼ í¬í•¨í•œë‹¤ [ê·¸ë¦¼ 4]. 

================================================================================
SAM file has at least 11 number of fields

SAM file contains following information
  - name of read
  - bitwise flags: alignment result of reads "into reference sequence"
  - name of reference sequence
  - locations of alignment
  - accuracy of alignment (or mapping quality)
  - CIGAR

================================================================================
SAM/BAM file example

/home/young/Pictures/2019_06_07_12:16:19.png

================================================================================
/ ì •ë ¬ ì—¬ë¶€ì— ëŒ€í•œ ì¤‘ìš”í•œ ì •ë³´ëŠ” bitwise flagsì— í¬í•¨ë˜ì–´ ìˆëŠ”ë°, ëª¨ë“  /
/ ë¦¬ë“œê°€ ê¸°ì¤€ ì—¼ê¸°ì„œì—´ì— ìœ ì¼í•˜ê³  ì •í™•í•˜ê²Œ ì •ë ¬ë˜ëŠ” ê²ƒì€ ì•„ë‹˜ìœ¼ë¡œ ë‹¤ì–‘í•œ ìœ í˜•ì˜ ì •ë ¬ ê²°ê³¼ê°€ /
/ ê°€ëŠ¥í•˜ê³  ì¤‘ìš”í•œ ìœ í˜•ì„ í‘œ 3ì— ìš”ì•½í•˜ì˜€ë‹¤. / 

================================================================================
Important information about the alignment
  - bigwise flags
  - All reads are not always aligned to "reference sequence" uniquely and precisely
  - it means there can be various types of alignment

================================================================================
Most important 3 types of alignment

/home/young/Pictures/2019_06_07_12:24:18.png

Type1: read are uniquely mapped to reference sequence
When: accuracy of read's sequence is high

Type2: read are ambiguously mapped to reference sequence
When: reads are mapped to "repetitive sequence" of "reference sequence"
when "reference sequence" comes from "homologuous region"

Type3: read are not mapped to reference sequence
When: sample for reads contains "chromosome" of other species
when reads contain "adapter"
when sequence in reads had been ambiguously sequenced
Error from PCR step
when reads are affected by structural variation

================================================================================
CIGARëŠ” ì •ë ¬ëœ ì„œì—´ì— ëŒ€í•œ ì •ë³´ë¥¼ ë¶€í˜¸í™”í•œ ê²ƒìœ¼ë¡œ,
ì •ë ¬ëœ ì—¼ê¸°ì˜ ê°œìˆ˜, insertion, deletion, skipped regionì— ê´€í•œ ì •ë³´ ë“±ì„ í‘œí˜„í•œë‹¤.

================================================================================
CIGAR:
  - encodes information of aligned sequence
  - contains information of 
  - number of aligned sequence (read)
  - insertion
  - deletion
  - skipped region

================================================================================
/ ë˜í•œ SAM /
/ íŒŒì¼ì„ ì´ì§„(binary) í˜•ì‹ìœ¼ë¡œ ì••ì¶•í•œ BAM íŒŒì¼ì€ SAM íŒŒì¼ê³¼ ë”ë¶ˆì–´ í˜„ì¬ ì •ë ¬ ê²°ê³¼ë¥¼ ì €ì¥í•˜ëŠ” /
/ í‘œì¤€í˜•ì‹ìœ¼ë¡œ ìë¦¬ì¡ê³  ìˆë‹¤. /

================================================================================
BAM_file = converter_into_binary_file(SAM_file)

SAM_file, BAM_file
  - Standard file format to save "result of alignment"

================================================================================
/ SAM/BAM íŒŒì¼ì„ íš¨ìœ¨ì ìœ¼ë¡œ ì²˜ë¦¬í•  ìˆ˜ ìˆëŠ” ì†Œí”„íŠ¸ì›¨ì–´ë¡œ SAMtools /
/ [26], GATK [27], Picard [28] ë“±ì´ í˜„ì¬ ë§ì´ ì‚¬ìš©ë˜ê³  ìˆë‹¤. BAM íŒŒì¼ì€ IGV(Integrative Genome /
/ Viewer) ì†Œí”„íŠ¸ì›¨ì–´ [29]ë¥¼ í†µí•´ ì‹œê°í™”í•  ìˆ˜ ìˆìœ¼ë©°, CRAMTools íŒ¨í‚¤ì§€ [30]ì„ ì‚¬ìš©í•˜ì—¬ CRAM /
/ í˜•ì‹ìœ¼ë¡œ ì••ì¶•í•  ìˆ˜ë„ ìˆë‹¤. /

================================================================================
Softwares to process SAM/BAM files
- SAMtools
- GATK
- Picard  

================================================================================
/ BAM íŒŒì¼ì€ IGV(Integrative Genome /
/ Viewer) ì†Œí”„íŠ¸ì›¨ì–´ [29]ë¥¼ í†µí•´ ì‹œê°í™”í•  ìˆ˜ ìˆìœ¼ë©°, CRAMTools íŒ¨í‚¤ì§€ [30]ì„ ì‚¬ìš©í•˜ì—¬ CRAM /
/ í˜•ì‹ìœ¼ë¡œ ì••ì¶•í•  ìˆ˜ë„ ìˆë‹¤. /

================================================================================
Visualization on BAM file
- IGV (Integrative Genome Viewer) software

Compress BAM file into CRAM file
- CRAMTools

================================================================================
/ BAM/SAM íŒŒì¼ì„ ìœ ì „ë³€ì´ ì¶”ì¶œ(variant calling) ë‹¨ê³„ì— ì ìš© í•˜ê¸° ì „ì— ì¼ë°˜ì ìœ¼ë¡œ ë‘ ê°€ì§€ /
/ ì„ í–‰ ì²˜ë¦¬, ì¦‰ ì •ë ¬ í›„ì† ê³¼ì •(alignment post processing)ê³¼ ì—¼ê¸° ì •í™•ë„ ì¬ë³´ì •(quality score /
/ recalibration)ì„ ìˆ˜í–‰í•œë‹¤. /

================================================================================
BAM/SAM
-> alignment post processing
-> quality score recalibration
-> variant calling (extracting genetic variant)

================================================================================
/ ì •ë ¬ í›„ì† ê³¼ì •ì€ ì •ë ¬ëœ ê²°ê³¼ë¥¼ ì—¼ìƒ‰ì²´ ë³„ë¡œ ë¶„ë¥˜í•˜ê³ , PCR ê³¼ì •ì—ì„œ /
/ ë°œìƒí•˜ëŠ” ì¸ê³µì  ê²°í•¨(artifact)ë¥¼ ì œê±°í•˜ë©°, ì°¸ì¡° ì—¼ê¸°ì„œì—´ì˜ ì—¬ëŸ¬ ìœ„ì¹˜ì— ë™ì‹œì— ì •ë ¬ëœ ë¦¬ë“œë“¤ì„ /
/ ì œê±°í•˜ëŠ” ê²ƒì„ í¬í•¨í•˜ê³ , Indelì„ ì¤‘ì‹¬ìœ¼ë¡œ ì§€ì—­ì  ì¬ì •ë ¬(local realignment)ì„ ì‹¤í–‰í•˜ê¸°ë„ í•œë‹¤. /
/ ì •ë ¬ í›„ì† ê³¼ì •ì€ ì†Œí”„íŠ¸ì›¨ì–´ SAMtools í˜¹ì€ Picard [í‘œ 4]ë“±ìœ¼ë¡œ ì²˜ë¦¬í•  ìˆ˜ ìˆë‹¤. /

================================================================================
alignment post processing

- Classify "aligned sequence of read" into "chromosomes"
- Remove artifacts which are created in PCR
- Remove "reads" which are aligned to multiple locations in "reference sequence"
- Perform "local realignment" with considering "Indel"
- Software
  - SAMtools
  - Picard

/home/young/Pictures/2019_06_07_12:49:56.png

================================================================================
/ SOAPsnpëŠ” /
/ ì‹œí€¸ì‹± í”Œë«í¼ì´ ì œê³µí•œ ì—¼ê¸° ì •í™•ë„, ë¦¬ë“œì—ì„œ ì—¼ê¸°ì˜ ìœ„ì¹˜(sequencing cycle), ëŒ€ì²´ /
/ ìœ í˜•(substitution type, ì˜ˆ: ì°¸ì¡° ì—¼ê¸°ì„œì—´ì—ì„œëŠ” A ì¸ë° ë¦¬ë“œì—ì„œëŠ” Gì¸ ê²½ìš°) ë“±ì„ ì‚¬ìš©í•˜ì—¬ ì°¸ì¡° /
/ ì—¼ê¸°ì„œì—´ì— ëŒ€í•˜ì—¬ í‰ê·  ë¶ˆì¼ì¹˜ ë¹„ìœ¨(mismatch rate)ë¥¼ ê³„ì‚°í•˜ê³ , ê·¸ ê²°ê³¼ë¥¼ ë³´ì •ëœ ì—¼ê¸° /
/ ì •í™•ë„(recalibrated quality score)ë¥¼ ê²°ì •í•˜ëŠ”ë° ì‚¬ìš©í•œë‹¤. /

================================================================================
quality score recalibration by SOAPsnp software

- Use following infromation to calculate "average mismatch rate"
between "read" and "reference sequence"

- Accuracy of sequence which is provided from "sequencing platform"
- locations of base in read (or sequencing cycle)
- substitution type
  - for example, case where A in reference sequence, but G in read
- Calculate result is used to determine "recalibrated quality score"

================================================================================
/ GATKë„ ìœ ì‚¬í•œ ë°©ë²•ì„ ì‚¬ìš©í•˜ëŠ”ë°, ì—¼ê¸°ë¥¼ /
/ ì •í™•ë„ì™€ dinucleotide content ë“±ìœ¼ë¡œ ì‹¤í—˜ì ì¸ ì˜¤ë¥˜ ëª¨ë¸ì„ ì‚¬ìš©í•˜ì—¬ ë¶„ë¥˜í•œ í›„, ê° ë¶„ë¥˜ êµ°ì— /
/ ëŒ€í•˜ì—¬ ì‹¤í—˜ì ì¸ ë¶ˆì¼ì¹˜ ë¹„ìœ¨ì„ ê³„ì‚°í•˜ì—¬ ë³´ì •ëœ ì—¼ê¸° ì •í™•ë„ë¥¼ ê²°ì •í•˜ëŠ” ìë£Œë¡œ ì‚¬ìš©í•œë‹¤. /

================================================================================
quality score recalibration by GATK software

- Classify "base (or sequence)" 
by using "accuracy" and "dinucleotide content"
- Calculate "mismatching ratio" with respect to each class
- Calculated ratio is used to determine accuracy of quality score recalibration

================================================================================
/ ì •ë ¬ê³¼ í›„ì† ê³¼ì •ì„ ë§ˆì¹œ ì—¼ê¸°ì„œì—´ ë°ì´í„°ì˜ íŠ¹ì„±ì€ ë„ˆë¹„(breath)ì™€ ê¹Šì´(depth í˜¹ì€ /
/ coverage) ë“± í¬ê²Œ ë‘ ê°€ì§€ ì²™ë„ë¡œ í‘œí˜„í•  ìˆ˜ ìˆë‹¤. ë„ˆë¹„ë€ ê·œëª…í•œ ìœ ì „ì²´ ì •ë„ë¥¼ ë‚˜íƒ€ë‚´ë©°, ê¹Šì´ë€ /
/ ìœ ì „ì²´ì—ì„œ ê° ì—¼ê¸°ê°€ ë¦¬ë“œì— ì˜í•´ í‰ê· ì ìœ¼ë¡œ ê·œëª…ëœ ì •ë„ë¥¼ ë‚˜íƒ€ë‚¸ë‹¤. ê¹Šì´ëŠ” ì—¼ê¸°ì— ëŒ€í•˜ì—¬ /
/ ëŒ€ëµì ìœ¼ë¡œ ì •ê·œë¶„í¬ë¥¼ ë”°ë¥¸ë‹¤. /

================================================================================
After, "alignment" and "postprocessing (alignment post processing quality score recalibration)"

sequence data can be expressed in "breath" and "depth (or coverage)"

Depth follows "Normal distribution" when "base" is variable

================================================================================
/ ë³€ì´ ì¶”ì¶œ(variant calling)ì´ë€ ì •ë ¬, ì •ë ¬ í›„ì† ê³¼ì •, ê·¸ë¦¬ê³  ì—¼ê¸° ì •í™•ë„ ì¬ë³´ì • ë“±ì„ ê±°ì¹œ /
/ ë¦¬ë“œë“¤ì˜ BAM/SAM íŒŒì¼ë“¤ì„ í†µí•©í•˜ì—¬ SNP/Indel ì˜ì—­ì„ ì°¾ëŠ” ê³¼ì •ì´ë‹¤. /

================================================================================
Variation calling (extracting genetic variant from read)

alignment
-> postprocessing after alignment
-> quality score recalibration
-> variation calling

variation calling
  - Merge BAM/SAM files of reads
  - Find regions of SNP/Indel

================================================================================
/ ë³€ì´ ì¶”ì¶œì€ ì—¼ê¸°ì„œì—´ì˜ /
/ ìœ„ì¹˜ë¥¼ SNP/Indelë¡œ í™•ì¸ë  í™•ë¥ ë¡œ í‘œí˜„í•˜ë©°, ì£¼ë¡œ ë² ì´ì§€ì•ˆ ë°©ë²•(Bayesian method) [25]ìœ¼ë¡œ /
/ ê³„ì‚°í•œë‹¤. /

================================================================================
variation calling expresses location of sequence 
as probability of "that sequence" confirmed as SNP/Indel 

Calculation:
  - Bayesian method

================================================================================
/ ë² ì´ì§€ì•ˆ ë°©ë²•ì´ë€ ì‚¬ì „í™•ë¥ (prior probability)ê³¼ ìš°ë„(likelihood)ë¥¼ ì‚¬ìš©í•˜ì—¬ /
/ ì‚¬í›„í™•ë¥ (posterior probability)ì„ ê³„ì‚°í•˜ëŠ” ë°©ë²•ìœ¼ë¡œ, ë‹¨ìˆœíˆ ìš°ë„ë§Œ ì‚¬ìš©í•˜ì—¬ ì¶”ì •í•˜ëŠ” ê²ƒë³´ë‹¤ ë” /
/ ë§ì€ ì •ë³´ë¥¼ ì–»ì„ ìˆ˜ ìˆë‹¤. /

================================================================================
Bayesian method:
  - Calculate posterior probability
  - by using prior probability and likelihood
  - By using Bayesian method,
  - you can get more information
  - than the method where you use only likelihood

================================================================================
/ ì´ë•Œ ì‚¬ì „í™•ë¥ ì€ ì„ í—˜ì ìœ¼ë¡œ ê° ì—¼ê¸° ìœ„ì¹˜(base position)ê°€ ìœ ì „ë³€ì´ê°€ /
/ ë  í™•ë¥ ì´ë©°, ìš°ë„ëŠ” ê·¸ ìœ„ì¹˜ì˜ ì—¼ê¸°ê°€ ì„œë¡œ ë‹¤ë¥¸ ìœ ì „ìí˜•ì´ ë  ë¹ˆë„(frequency)ì´ë‹¤. /

================================================================================
prior probability
  - probability of each base position becoming "genetic variation"

likelihood
  - frequency of specific location's base becoming differening genetic type

================================================================================
/ ì¦‰, ì–´ëŠ /
/ locusì—ì„œ ìƒ˜í”Œ ğ‘–ì˜ ìœ ì „ìí˜•ì´ ğºğ‘–ê°€ ë  í™•ë¥ ì€ NGS ë°ì´í„° ì§‘í•© ğ·ì™€ ë² ì´ì¦ˆ ê·œì¹™(Bayesâ€™ rule)ì— /
/ ì˜í•˜ì—¬ /

================================================================================
$$$p(G_i|D) 
= \dfrac{p(G_i) p(D|G_i)}{ \sum\limits_{k=1}^{S} p(G_k) p(D|G_k)}$$$

$$$D$$$: NGS data set
$$$i$$$: ith sample from locus
$$$G_i$$$: genetic type of sample i
$$$S$$$: number of genetic type
$$$p(G_i)$$$: prior probability
$$$p(D|G_i)$$$: likelihood

- prior probability $$$p(G_i)$$$ is predicted 
by using SNP database like dbSNP

================================================================================
/ ë³€ì´ ì¶”ì¶œ ê²°ê³¼ëŠ” VCF(variant call format) [32] í˜•ì‹ì˜ íŒŒì¼ì— ì €ì¥ëœë‹¤. /

================================================================================
Result from variation calling (extracting genetic variant from read)
  - is saved into VCF (variant call format) file

================================================================================
/ VCF íŒŒì¼ì€ /
/ SNP/Indelë¡œ í™•ì¸ëœ ìœ ì „ë³€ì´ í›„ë³´ì— ëŒ€í•œ ì •ë³´, ì¦‰ ì—¼ìƒ‰ì²´ ìœ„ì¹˜, ì°¸ì¡° ì—¼ê¸°, ì¹˜í™˜/ì‚½ì…/ì‚­ì œ ë“±ì´ /
/ ì¼ì–´ë‚œ í‘œë³¸ ì—¼ê¸°, ë³€ì´ ì •í™•ë„(variation quality), ë³€ì´ë¡œ íŒëª…ëœ ìƒ˜í”Œì˜ ë¹ˆë„ ë“±ì„ í¬í•¨í•˜ê³  ìˆë‹¤. /

================================================================================
VCF file
  - contains following information which is confirmed via SNP/Indel
  - information about candidate genetic variation
  - location of chromosome (of genetic variant occured?)
  - reference sequence
  - sample sequence which is occurred by
  - substitution, insertion, deletion
  - variation quality
  - frequency of sample which is confirmed as genetic variation

================================================================================
/ ë˜í•œ ë³€ì´ ë¹ˆë„(variation frequency)ì— ì˜í•´ SNPëŠ” homozygousì™€ heterozygous ë“± ë‘ ê°€ì§€ /
/ ìœ í˜•ìœ¼ë¡œ ë‚˜ë‰œë‹¤. HomozygousëŠ” í•´ë‹¹ ìœ„ì¹˜ì— ì •ë ¬ëœ ê±°ì˜ ëª¨ë“  ìƒ˜í”Œì—ì„œ ë³€ì´ê°€ ì¼ì–´ë‚œ ê²½ìš°ì´ê³ , /
/ heterozygousëŠ” ì¼ë¶€ ìƒ˜í”Œì—ì„œë§Œ ë³€ì´ê°€ ì¼ì–´ë‚œ ê²½ìš°ì´ë‹¤. ê·¸ë¦¼ 5ëŠ” SAMtoolsë¥¼ ì‚¬ìš©í•˜ì—¬ êµ¬í•œ /
/ VCF í˜•ì‹ì˜ íŒŒì¼ì´ë‹¤. /

================================================================================
SNP:
  - has following 2 types due to variation frequency
  - homozygous
  - heterozygous

================================================================================
/ HomozygousëŠ” í•´ë‹¹ ìœ„ì¹˜ì— ì •ë ¬ëœ ê±°ì˜ ëª¨ë“  ìƒ˜í”Œì—ì„œ ë³€ì´ê°€ ì¼ì–´ë‚œ ê²½ìš°ì´ê³ , /
/ heterozygousëŠ” ì¼ë¶€ ìƒ˜í”Œì—ì„œë§Œ ë³€ì´ê°€ ì¼ì–´ë‚œ ê²½ìš°ì´ë‹¤. ê·¸ë¦¼ 5ëŠ” SAMtoolsë¥¼ ì‚¬ìš©í•˜ì—¬ êµ¬í•œ /
/ VCF í˜•ì‹ì˜ íŒŒì¼ì´ë‹¤.

================================================================================
Homozygous:
  - Almost all samples which are aligned to reference sequence
  have genetic variation

heterozygous:
  - Only some part of samples which are aligned to reference sequence
  have genetic variation

================================================================================
/ ê·¸ë¦¼ 5ëŠ” SAMtoolsë¥¼ ì‚¬ìš©í•˜ì—¬ êµ¬í•œ /
/ VCF í˜•ì‹ì˜ íŒŒì¼ì´ë‹¤. /

================================================================================
VCF file example which is obtained via SAMtools

/home/young/Pictures/2019_06_07_13:56:45.png

================================================================================
/ ë³€ì´ ì¶”ì¶œì„ í†µí•´ ì–»ì–´ì§„ SNP/Indelì€ ì ì¬ì ì¸ ê²ƒìœ¼ë¡œ ëª¨ë‘ ìœ ì˜í•œ ìœ ì „ë³€ì´ë¼ê³  ê°„ì£¼í•˜ê¸° /
/ ì–´ë µê¸° ë•Œë¬¸ì— í•„í„°ë§ì„ í†µí•´ ê±°ì§“ì–‘ì„±(false-positive)ì— í•´ë‹¹í•˜ëŠ” SNP/Indelì„ ì œê±°í•˜ê³  ìœ ì˜í•œ /
/ SNP/Indelë§Œ ê³¨ë¼ë‚´ëŠ” ê³¼ì •ì„ ê±°ì¹œë‹¤. ë³´í†µ ì‚¬ìš©í•˜ëŠ” ë°©ë²•ì€ Hardy-Weinberg í‰í˜•ì—ì„œ ë²—ì–´ë‚œ ì •ë„, /
/ ë¦¬ë“œì˜ ìµœëŒ€ ë° ìµœì†Œ ê¹Šì´, ì¸ì ‘í•œ Indels ë“±ì„ ê¸°ì¤€ìœ¼ë¡œ ì¡°ì‚¬í•œë‹¤. /

================================================================================
- You obtain SNP/Indel from "variation calling step"
- But you shouldn't consider all SNP/Indel as significant genetic variation
- So, you should filter false-positive SNP/Indel
- to have only significant SNP/Indel
- Method:
  - Detect SNP/Indel which is out of Hardy-Weinberg equilibrium state
  - Inspect maximal/minimal depth of reads
  - Inspect neighboring Indels
  - With them, you can classify significant SNL/Indel

================================================================================
/ íŠ¹ë³„íˆ ì •í•´ì ¸ ìˆëŠ” ê¸°ì¤€ì´ ì—†ê¸° /
/ ë•Œë¬¸ì— ì •ë ¬ëœ ë¦¬ë“œë“¤ì˜ ê¹Šì´, ë³€ì´ë¡œ íŒëª…ëœ ìƒ˜í”Œì˜ ë¹ˆë„ ë“±ì„ ê³ ë ¤í•œ ì ì • ê¸°ì¤€ì„ ì„¸ì›Œ í•„í„°ë§ì„ /
/ ì§„í–‰í•œë‹¤. í•„í„°ë§ì„ ìœ„í•œ ì†Œí”„íŠ¸ì›¨ì–´ë¡œ GATKì˜ VariantFiltration, vcfutil.pl, SAMtools, ê·¸ë¦¬ê³  /
/ VCFtools [32] ë“±ì´ ìˆë‹¤. íŠ¹íˆ VCFtoolsì€ VCF íŒŒì¼ì„ ë‹¤ì–‘í•œ ë°©ë©´ìœ¼ë¡œ ë‹¤ë£¨ëŠ” ì—­í• ì„ í•  ìˆ˜ /
/ ìˆëŠ”ë°, ì—¬ëŸ¬ íŒŒì¼ì„ í•©ì¹˜ê±°ë‚˜ í•„ìš”í•œ ì˜ì—­ì— ì†í•˜ëŠ” SNPë§Œ ì¶”ì¶œí•  ìˆ˜ ìˆë‹¤. /

================================================================================
Software for filtering SNP/Indel
  - GATK - VariantFiltration
  - vcfutil.pl
  - SAMtools
  - VCFtools

================================================================================
/ NGSë¡œ ìƒì„±í•œ FASTQ íŒŒì¼ì˜ í¬ê¸°ëŠ” ì••ì¶•ì„ í•´ë„ ìƒ˜í”Œ ë‹¹ ê¸°ê°€ ë°”ì´íŠ¸(gigabyte, 109 bytes) /
/ ìˆ˜ì¤€ì— ì´ë¥¸ë‹¤. BAM/SAM íŒŒì¼ê³¼ VCF íŒŒì¼ì˜ ìš©ëŸ‰ì„ ê³ ë ¤í•˜ë©´ ìƒ˜í”Œ 100ê°œì˜ íŒŒì¼ ìš©ëŸ‰ì€ í…Œë¼ /
/ ë°”ì´íŠ¸(terabyte, 1012bytes) ìˆ˜ì¤€ì´ ëœë‹¤.

================================================================================
FASTQ file (from NGS) size (after compression): GB per sample

FASTQ file + BAM/SAM + VCF file (from NGS): TB per 100 samples

================================================================================
/ Nvidia CUDAë¥¼ ì‚¬ìš©í•˜ì—¬ BWA /
/ ì•Œê³ ë¦¬ì¦˜ì„ ì ìš©í•œ ì •ë ¬ í”„ë¡œê·¸ë¨ì¸ Barracuda  /

================================================================================
BWA algorith with CUDA: Barracuda

================================================================================
/ 1990ë…„ëŒ€ê¹Œì§€ ê°œì¸ìš© ì»´í“¨í„°ì˜ ë©”ëª¨ë¦¬(random access memory, RAM)ëŠ” ì£¼ë¡œ 32-bits /
/ ìš´ì˜ì²´ì œì—ì„œ ì‚¬ìš©ë˜ì—ˆê¸° ë•Œë¬¸ì— ìš©ëŸ‰ì€ ìµœëŒ€ ì•½ 4GBê¹Œì§€ í—ˆìš©ë˜ì—ˆë‹¤. ë”°ë¼ì„œ ìš´ì˜ì²´ì œ ë° /
/ ì‹¤í–‰ì¤‘ì¸ ê¸°íƒ€ ì†Œí”„íŠ¸ì›¨ì–´ë¥¼ ìœ„í•œ ë©”ëª¨ë¦¬ë¥¼ í¬í•¨í•˜ì—¬ 4GB ì´ìƒì˜ ë©”ëª¨ë¦¬ë¥¼ ìš”êµ¬í•˜ëŠ” ìœ ì „ì²´ /
/ ë°ì´í„°ì˜ ë¶„ì„ì€ ë¶ˆê°€ëŠ¥í•˜ì˜€ë‹¤. 2004ë…„ê²½ 64-bits ì»´í“¨íŒ…ì„ ì§€ì›í•˜ëŠ” CPUê°€ ë“±ì¥í•¨ì— ë”°ë¼ ìš´ì˜ì²´ì œ /
/ ì—­ì‹œ ìµœëŒ€ ì•½ 8TBì˜ ë©”ëª¨ë¦¬ë¥¼ ì§€ì›í•  ìˆ˜ ìˆê¸° ë˜ì–´, NGS ë°ì´í„° ì „ì²´ë¥¼ ë©”ëª¨ë¦¬ì— íƒ‘ì¬í•˜ì—¬ ë¶„ì„í•  /
/ ìˆ˜ ìˆê²Œ ë˜ì—ˆë‹¤ /

32 bit OS allows 4GB maximal memory

64 bit OS allows 8TB maximal memory

================================================================================
/ NGSë¡œ ëŒ€ìš©ëŸ‰ì˜ ìœ ì „ì •ë³´ë¥¼ ìƒì‚°í•  ìˆ˜ ìˆìŒì— ë”°ë¼ ì¸ê°„ë¿ë§Œ ì•„ë‹ˆë¼ ë‹¤ì–‘í•œ ë™ì‹ë¬¼ì˜ /
/ ìœ ì „ì²´ í•´ë…ì´ ë³´í¸í™” ë˜ê³  ìˆìœ¼ë©°, ì§ˆë³‘ ì§„ë‹¨ ë° ì˜ˆì¸¡ê³¼ ìœ ìš© ìœ ì „ì •ë³´ ë°œêµ´ ë° ìœ¡ì¢…ì— ì‘ìš© ë“±ì´ /
/ ê°€ëŠ¥í•˜ê²Œ ë˜ì—ˆë‹¤. NGS ë°ì´í„°ì˜ ì¬ë¶„ì„ì„ í†µí•œ ìƒë¬¼í•™ì  í•´ì„ê³¼ í™œìš©ì€ ì›ì‹œ ë°ì´í„°ì˜ ë¶„ì„ ê²°ê³¼ì¸ /
/ SNP/Indel callingì˜ ì •í™•ë„ê°€ ì „ì œë˜ì–´ì•¼ í•˜ë©°, ì •í™•ë„ëŠ” ë¶„ì„ ë°©ë²•ë¡ ê³¼ ì•Œê³ ë¦¬ì¦˜ì— ì˜í–¥ì„ ë°›ëŠ”ë‹¤. /

================================================================================
NGS
  - can create massive genetic information data

Re-analysis on NGS data
  - can be used for biological interpretation
  - but integrity of "SNP/Indel calling" should be prepared for re-analysis on NGS data

================================================================================
/ ë”°ë¼ì„œ FASTQ íŒŒì¼ì—ì„œ SNP callingì— ì´ë¥´ëŠ” ë¶„ì„ íŒŒì´í”„ë¼ì¸ì— ëŒ€í•œ ì§€ì†ì ì¸ ì—°êµ¬ê°€ í•„ìš”í•˜ë©°, /
/ ìƒë¬¼í•™ì  ì—°êµ¬ ëª©ì ì— ë”°ë¼ ê¸°ëŠ¥ ì¶”ê°€/ì‚­ì œ ë“±ì„ í¬í•¨í•œ íŒŒì´í”„ë¼ì¸ì˜ êµ¬ì¡°ë¥¼ ë³€í™”ì‹œí‚¨ ë§ì¶¤í˜• /
/ íŒŒì´í”„ë¼ì¸ì„ êµ¬ì¶•í•  ìˆ˜ ìˆëŠ” ë°ì´í„° ë¶„ì„ ê¸°ìˆ ì´ ìš”êµ¬ëœë‹¤. /

================================================================================
Researches are needed for pipeline

- FASTQ file
...
- SNP calling

================================================================================
/ í˜„ì¬ëŠ” 1,000 genome /
/ projectì™€ ê°™ì€ ëŒ€í˜• ìœ ì „ì²´ í”„ë¡œì íŠ¸ì—ì„œ ì‚¬ìš©í•˜ëŠ” íŒŒì´í”„ë¼ì¸ì„ í‘œì¤€ìœ¼ë¡œ ê°„ì£¼í•˜ê³  ìˆìœ¼ë‚˜, ì•„ì§ /
/ ì˜ê²¬ ì¼ì¹˜ë¥¼ ë³´ì´ëŠ” ê²ƒì€ ì•„ë‹ˆë©°, ë˜í•œ íŒŒì´í”„ë¼ì¸ì´ ì›Œë‚™ ë³µì¡í•˜ì—¬ â€˜í‘œì¤€ íŒŒì´í”„ë¼ì¸â€™ì´ ê°€ëŠ¥í•œì§€ë„ /
/ í˜„ì¬ë¡œëŠ” ì˜ë¬¸ì´ë‹¤. /

================================================================================
Pipeline which is used for "1,000 genome project" is considered as standard

================================================================================
/ í•œí¸ â€œThe Genome in a Bottle Consortiumâ€ [37, 38]ì—ì„œ DNA ì‹œí€€ì‹±ì˜ í‘œì¤€ /
/ ë°ì´í„°ë¥¼ ì œê³µí•˜ê³  ìˆìŒìœ¼ë¡œ, í–¥ í›„ ë‹¤ì–‘í•œ ì†Œí”„íŠ¸ì›¨ì–´ ë° íŒŒì´í”„ë¼ì¸ì„ ë¹„êµ ê²€í† í•˜ëŠ”ë° ì‚¬ìš©í•  ìˆ˜ /
/ ìˆì„ ê²ƒìœ¼ë¡œ ê¸°ëŒ€ëœë‹¤ /

================================================================================
The Genome in a Bottle Consortium provides standard data for DNA sequencing

================================================================================


</xmp>
   </BODY>
</HTML>
